docs/design/proposals/status-conditions.md

```diff
@@ -0,0 +1,254 @@
+
+The condition manager is responsible for propagating the most important and most
+recent change to the top-level condition.
+
+Another side effect of using a condition manager is that sub-conditions become
+implementation details and the only thing that is required for an observer to
+observe is the top-level condition.
+
+##### On Condition Status
+
+There are three and only three values supported by a condition.Status: `True`,
+`False`, and `Unknown`.
+
+- True means the condition passes and there are no concerns for an operator. All
+  should be well and working. In happy systems, things are all true.
+- False means there is a fundamental issue that requires intervention. The
+  reconciler is stopped from making progress until that issue is resolved. This
+  could be as simple as an incorrect schema, or as complex as a downstream
+  system only returning 500s and some on-call needs to be called.
+- Unknown means the reconciler is making progress, but has not yet successfully
+  fulfilled the request.
+
+> Note: A False condition is nearly as serious as a fatal log line. It is not to
+> be used lightly.
+
+Resource Graph DefinitionThe Resource Graph Definition currently uses 3
+conditions:
+
+- GraphVerified
+- CustomResourceDefinitionSynced
+- ReconcilerReady
+
+A good start but we violate the principle of a standard top-level condition,
+Ready. This should be added.
+
+GraphVerified is hard to link to real code at the moment because the setting of
+this value is abstracted.
+
+CustomResourceDefinitionSynced is not a great name because what we really want
+to talk about is all of the tasks around creating, updating, and watching the
+CRD is accepted. I suspect we don’t watch the CRD very closely as I see cases
+where I can delete the CRD KRO has made for me and the controller does not bring
+it back.
+
+##### Example
+
+The reconciler would be updated to remove the setConditions methods and replaced
+with something more like (these are just off the cuff examples):
+
+```
+rdg.Status.PropagateCRDStatus(currentCRD)
+rgd.Status.MarkCRDFailed(errorMessage)
+
+rdg.Status.PropagateResourceGraphState(listOfResources)
+rgd.Status.MarkFailedResourceUpdate(errorMessage)
+```
+
+Status would be updated as follows:
+
+`Current RGD Status`
+
+```
+status:
+  conditions:
+  - lastTransitionTime: "2025-03-24T21:31:02Z"
+    message: micro controller is ready
+    reason: ""
+    status: "True"
+    type: ReconcilerReady
+  - lastTransitionTime: "2025-03-24T21:31:02Z"
+    message: Directed Acyclic Graph is synced
+    reason: ""
+    status: "True"
+    type: GraphVerified
+  - lastTransitionTime: "2025-03-24T21:31:02Z"
+    message: Custom Resource Definition is synced
+    reason: ""
+    status: "True"
+    type: CustomResourceDefinitionSynced
+```
+
+`Proposed Update`
+
+```
+status:
+  conditions:
+  - lastTransitionTime: "2025-03-24T21:31:02Z"
+    message: ""
+    observedGeneration: 1
+    reason: "AllReady"
```

Comment:
ResourcesReady? This corresponds to the `resources` key in the RGD.

Reply:
here is what got implemented, this document is a pure port of the design doc from a few months ago.